Mathlib/LinearAlgebra/Pi.lean

```diff
@@ -113,25 +114,105 @@ theorem apply_single [AddCommMonoid M] [Module R M] [DecidableEq ι] (f : (i : 
     (i j : ι) (x : φ i) : f j (Pi.single i x j) = (Pi.single i (f i x) : ι → M) j :=
   Pi.apply_single (fun i => f i) (fun i => (f i).map_zero) _ _ _
 
+variable (R φ)
+
 /-- The `LinearMap` version of `AddMonoidHom.single` and `Pi.single`. -/
 def single [DecidableEq ι] (i : ι) : φ i →ₗ[R] (i : ι) → φ i :=
   { AddMonoidHom.single φ i with
     toFun := Pi.single i
     map_smul' := Pi.single_smul i }
 
+lemma single_apply [DecidableEq ι] {i : ι} (v : φ i) :
+    single R φ i v = update (0 : (a : ι) → φ a) i v :=
+  rfl
+
+lemma single_apply' [DecidableEq ι] (i i' : ι) :
+    single R (fun _ ↦ R) i 1 i' = if (i = i') then 1 else 0  := by
+  rw [single_apply, update_apply]
+  congr 1; rw [eq_iff_iff, eq_comm]
+
 @[simp]
-theorem coe_single [DecidableEq ι] (i : ι) : ⇑(single i : φ i →ₗ[R] (i : ι) → φ i) = Pi.single i :=
+theorem coe_single [DecidableEq ι] (i : ι) :
+    ⇑(single R φ i : φ i →ₗ[R] (i : ι) → φ i) = Pi.single i :=
   rfl
 
-variable (R φ)
+variable [DecidableEq ι]
+
+@[simp]
+theorem single_same (i : ι) (b : φ i) : single R φ i b i = b :=
+  Pi.single_eq_same i b
+
+theorem single_ne (i j : ι) (h : j ≠ i) (b : φ i) : single R φ i b j = 0 :=
+  Pi.single_eq_of_ne h b
```

Comment:
I'd propose we delete these and `single_apply'` and `single_apply` too, and poitn the deprecations directly at the `Pi.single` lemmas. I'll push a change to that effect once everything else is passing.

Reply:
(I did this)